src/textinput.rs

```diff
@@ -61,16 +157,161 @@ impl FromXml for TextInput {
 
         writer.write(Event::End(element))
     }
 }
+
+/// This `TextInputBuilder` struct creates the `TextInput`.
+#[derive(Debug, Clone, Default)]
+pub struct TextInputBuilder {
+    title: String,
+    description: String,
+    name: String,
+    link: String,
+}
+
+impl TextInputBuilder {
+    /// Construct a new `TextInputBuilder` and return default values.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::TextInputBuilder;
+    ///
+    /// let text_input_builder = TextInputBuilder::new();
+    /// ```
+    pub fn new() -> TextInputBuilder {
+        TextInputBuilder::default()
+    }
+
+    /// Set the title that exists under `TextInput`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::TextInputBuilder;
+    ///
+    /// let mut text_input_builder = TextInputBuilder::new();
+    /// text_input_builder.title("Title");
+    /// ```
+    pub fn title(mut self,
+                 title: &str)
+        -> TextInputBuilder {
+        self.title = title.to_owned();
+        self
+    }
+
+    /// Set the description that exists under `TextInput`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::TextInputBuilder;
+    ///
+    /// let mut text_input_builder = TextInputBuilder::new();
+    /// text_input_builder.description("This is a test description.");
+    /// ```
+    pub fn description(mut self,
+                       description: &str)
+        -> TextInputBuilder {
+        self.description = description.to_owned();
+        self
+    }
+
+    /// Set the name that exists under `TextInput`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::TextInputBuilder;
+    ///
+    /// let mut text_input_builder = TextInputBuilder::new();
+    /// text_input_builder.name("Comments");
+    /// ```
+    pub fn name(mut self,
+                name: &str)
+        -> TextInputBuilder {
+        self.name = name.to_owned();
+        self
+    }
+
+    /// Set the link that exists under `TextInput`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::TextInputBuilder;
+    ///
+    /// let mut text_input_builder = TextInputBuilder::new();
+    /// text_input_builder.link("http://www.example.com/feedback");
+    /// ```
+    pub fn link(mut self,
+                link: &str)
+        -> TextInputBuilder {
+        self.link = link.to_owned();
+        self
+    }
+
+    /// Validate the contents of `TextInput`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rss::TextInputBuilder;
+    ///
+    /// let text_input = TextInputBuilder::new()
+    ///         .title("Title")
+    ///         .description("This is a test description.")
+    ///         .name("Comments")
+    ///         .link("http://www.example.com/feedback")
+    ///         .validate()
+    ///         .unwrap()
+    ///         .finalize()
+    ///         .unwrap();
+    /// ```
+    pub fn validate(self) -> Result<TextInputBuilder, Error> {
+        Url::parse(self.link
```

Comment:
Is this `clone` required? I looks like it would work the same with just calling `as_str()`.